apis/sgv2-graphqlapi/src/main/java/io/stargate/sgv2/graphql/web/resources/GraphqlResourceBase.java

```diff
@@ -232,39 +265,68 @@ public void bindFilesToVariables(GraphqlFormData formData) {
+                            "You must provide a GraphQL query in the request body"));
+              }
 
-    ExecutionInput input = ExecutionInput.newExecutionInput(query).context(context).build();
-    executeAsync(input, graphql, asyncResponse);
+              ExecutionInput input =
+                  ExecutionInput.newExecutionInput(query).context(context).build();
+              return execute(input, graphql);
+            })
+        // map to rest response
+        .map(RestResponse::ok);
   }
 
-  protected static void executeAsync(
-      ExecutionInput input, GraphQL graphql, @Suspended AsyncResponse asyncResponse) {
-    graphql
-        .executeAsync(input)
-        .whenComplete(
-            (result, error) -> {
-              if (error != null) {
-                LOG.error("Unexpected error while processing GraphQL request", error);
-                throw graphqlError(Response.Status.INTERNAL_SERVER_ERROR, "Internal server error");
+  protected static Uni<Map<String, Object>> execute(ExecutionInput input, GraphQL graphql) {
+    // execute graphql call
+    return executeGraphql(input, graphql)
+
+        // on item check if we are not maybe overloaded
+        .onItem()
+        .transformToUni(
+            result -> {
+              Object context = input.getContext();
+              if (context instanceof StargateGraphqlContext
+                  && ((StargateGraphqlContext) context).isOverloaded()) {
+                return Uni.createFrom()
+                    .failure(
+                        graphqlError(Response.Status.TOO_MANY_REQUESTS, "Database is overloaded"));
               } else {
-                Object context = input.getContext();
-                if (context instanceof StargateGraphqlContext
-                    && ((StargateGraphqlContext) context).isOverloaded()) {
-                  throw graphqlError(Response.Status.TOO_MANY_REQUESTS, "Database is overloaded");
-                } else {
-                  asyncResponse.resume(result.toSpecification());
-                }
+                return Uni.createFrom().item(result.toSpecification());
               }
+            })
+
+        // on failure map to web app exception
+        .onFailure()
+        .recoverWithUni(
+            error -> {
+              LOG.error("Unexpected error while processing GraphQL request", error);
+              return Uni.createFrom()
+                  .failure(
+                      graphqlError(Response.Status.INTERNAL_SERVER_ERROR, "Internal server error"));
             });
   }
 
+  private static Uni<ExecutionResult> executeGraphql(ExecutionInput input, GraphQL graphql) {
+    // create uni from future
+    return Uni.createFrom()
+        .future(() -> graphql.executeAsync(input))
+
+        // always run subscription on workers thread
+        // b/c although return type is completable future
+        // we are blocking inside of the graphql
+        .runSubscriptionOn(Infrastructure.getDefaultWorkerPool());
```

Comment:
Yes exactly.. If we would refactor the fetchers, we would be able to remove this subscription on the workers pool and run everything directly on the vertx thread pool..

I already tried to see how much effort this would be, but it is not small.. However, I am all about doing it at some point..